pkg/apis/extensions/v1beta1/types.go

```diff
@@ -516,10 +546,16 @@ type DaemonSet struct {
 }
 
 const (
+	// DEPRECATED: DefaultDaemonSetUniqueLabelKey is used instead.
 	// DaemonSetTemplateGenerationKey is the key of the labels that is added
 	// to daemon set pods to distinguish between old and new pod templates
 	// during DaemonSet template update.
 	DaemonSetTemplateGenerationKey string = "pod-template-generation"
+
+	// DefaultDaemonSetUniqueLabelKey is the default label key that is added
+	// to existing DaemonSet pods to distinguish between old and new
+	// DaemonSet pods during DaemonSet template updates.
+	DefaultDaemonSetUniqueLabelKey string = "daemonset-controller-revision"
```

Comment:
Not using `pod-template-hash` intentionally so that we differentiate histories from different controllers. 

No need to relabel, each history will have a fixed hash label and the label is used by DS to compare the DS's template with the history and to label DS's pods